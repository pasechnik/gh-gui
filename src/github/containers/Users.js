import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { AsyncTypeahead } from 'react-bootstrap-typeahead'
import {
  FormGroup, Label, Row, Col,
} from 'reactstrap'
import {
  doUsers as fUsers, doLoginSet as fLoginSet, doUserSet as fUserSet,
} from '../actions/doUsers'

import 'react-bootstrap-typeahead/css/Typeahead-bs4.css'

export const Users = (
  {
    users, login, loading, doUsers, doLoginSet, doUserSet,
  },
) => (
  <Row>
    <Col md={2}>
      <Label for='login'>
        User login
      </Label>
    </Col>
    <Col md={10}>
      <FormGroup>
        <AsyncTypeahead
          id='inputsearch'
          isLoading={loading}
          labelKey='login'
          minLength={3}
          onSearch={doUsers}
          placeholder='Search for a Github user...'
          onChange={doUserSet}
          onInputChange={doLoginSet}
          options={users}
          defaultInputValue={login}
          // renderMenuItemChildren={(option, props) => (<li key={option.id} user={option} />)}
        />
      </FormGroup>
    </Col>
  </Row>
)

Users.propTypes = {
  loading: PropTypes.bool.isRequired,
  users: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  login: PropTypes.string.isRequired,
  doUsers: PropTypes.func.isRequired,
  doLoginSet: PropTypes.func.isRequired,
  doUserSet: PropTypes.func.isRequired,
}

const mapStateToProps = state => ({
  loading: state.github.users.isLoading,
  users: state.github.users.data,
  login: state.github.users.login,
  user: state.github.users.user,
})

const mapDispatchToProps = {
  doUsers: fUsers,
  doLoginSet: fLoginSet,
  doUserSet: fUserSet,
}

export default connect(mapStateToProps, mapDispatchToProps)(Users)
