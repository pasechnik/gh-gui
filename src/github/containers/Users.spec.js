import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import React from 'react'

import { Users } from './Users'

configure({ adapter: new Adapter() })

const setup = () => {
  const props = {
    loading: false,
    users: [],
    login: 'test',
    doUsers: jest.fn(),
    doLoginSet: jest.fn(),
    doUserSet: jest.fn(),
  }

  const wrapper = shallow(<Users {...props} />)

  return {
    props,
    wrapper,
  }
}

describe('<Users />', () => {
  it('Renders', () => {
    const {
      wrapper,
    } = setup()

    expect(wrapper).toMatchSnapshot()
  })

  // Doesn't work with AsyncTypeahead
  // it('should call search users if length of text is greater than 3', () => {
  //   const {
  //     props,
  //     wrapper,
  //   } = setup()
  //
  //   const input = wrapper.find('#inputsearch')
  //   input.simulate('focus')
  //   input.simulate('change', { target: { value: 'te' } })
  //   expect(props.doLoginSet.mock.calls.length).toBe(0)
  //   input.simulate('change', { target: { value: 'testusche' } })
  //   expect(props.doLoginSet.mock.calls.length).toBe(1)
  // })
})
