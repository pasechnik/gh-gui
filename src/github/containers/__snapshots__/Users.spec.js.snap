// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Users /> Renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    doLoginSet={[Function]}
    doUserSet={[Function]}
    doUsers={[Function]}
    loading={false}
    login="test"
    users={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Col
          md={2}
          tag="div"
          widths={
                    Array [
                              "xs",
                              "sm",
                              "md",
                              "lg",
                              "xl",
                            ]
          }
>
          <Label
                    for="login"
                    tag="label"
                    widths={
                              Array [
                                        "xs",
                                        "sm",
                                        "md",
                                        "lg",
                                        "xl",
                                      ]
                    }
          >
                    User login
          </Label>
</Col>,
        <Col
          md={10}
          tag="div"
          widths={
                    Array [
                              "xs",
                              "sm",
                              "md",
                              "lg",
                              "xl",
                            ]
          }
>
          <FormGroup
                    tag="div"
          >
                    <AsyncContainer(OnClickOutside(TypeaheadContainer(WrappedTypeahead)))
                              defaultInputValue="test"
                              delay={200}
                              id="inputsearch"
                              isLoading={false}
                              labelKey="login"
                              minLength={3}
                              onChange={[Function]}
                              onInputChange={[Function]}
                              onSearch={[Function]}
                              options={Array []}
                              placeholder="Search for a Github user..."
                              promptText="Type to search..."
                              searchText="Searching..."
                              useCache={true}
                    />
          </FormGroup>
</Col>,
      ],
      "tag": "div",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Label
            for="login"
            tag="label"
            widths={
                        Array [
                                    "xs",
                                    "sm",
                                    "md",
                                    "lg",
                                    "xl",
                                  ]
            }
>
            User login
</Label>,
          "md": 2,
          "tag": "div",
          "widths": Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "User login",
            "for": "login",
            "tag": "label",
            "widths": Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ],
          },
          "ref": null,
          "rendered": "User login",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <FormGroup
            tag="div"
>
            <AsyncContainer(OnClickOutside(TypeaheadContainer(WrappedTypeahead)))
                        defaultInputValue="test"
                        delay={200}
                        id="inputsearch"
                        isLoading={false}
                        labelKey="login"
                        minLength={3}
                        onChange={[Function]}
                        onInputChange={[Function]}
                        onSearch={[Function]}
                        options={Array []}
                        placeholder="Search for a Github user..."
                        promptText="Type to search..."
                        searchText="Searching..."
                        useCache={true}
            />
</FormGroup>,
          "md": 10,
          "tag": "div",
          "widths": Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <AsyncContainer(OnClickOutside(TypeaheadContainer(WrappedTypeahead)))
              defaultInputValue="test"
              delay={200}
              id="inputsearch"
              isLoading={false}
              labelKey="login"
              minLength={3}
              onChange={[Function]}
              onInputChange={[Function]}
              onSearch={[Function]}
              options={Array []}
              placeholder="Search for a Github user..."
              promptText="Type to search..."
              searchText="Searching..."
              useCache={true}
/>,
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "defaultInputValue": "test",
              "delay": 200,
              "id": "inputsearch",
              "isLoading": false,
              "labelKey": "login",
              "minLength": 3,
              "onChange": [Function],
              "onInputChange": [Function],
              "onSearch": [Function],
              "options": Array [],
              "placeholder": "Search for a Github user...",
              "promptText": "Type to search...",
              "searchText": "Searching...",
              "useCache": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Col
            md={2}
            tag="div"
            widths={
                        Array [
                                    "xs",
                                    "sm",
                                    "md",
                                    "lg",
                                    "xl",
                                  ]
            }
>
            <Label
                        for="login"
                        tag="label"
                        widths={
                                    Array [
                                                "xs",
                                                "sm",
                                                "md",
                                                "lg",
                                                "xl",
                                              ]
                        }
            >
                        User login
            </Label>
</Col>,
          <Col
            md={10}
            tag="div"
            widths={
                        Array [
                                    "xs",
                                    "sm",
                                    "md",
                                    "lg",
                                    "xl",
                                  ]
            }
>
            <FormGroup
                        tag="div"
            >
                        <AsyncContainer(OnClickOutside(TypeaheadContainer(WrappedTypeahead)))
                                    defaultInputValue="test"
                                    delay={200}
                                    id="inputsearch"
                                    isLoading={false}
                                    labelKey="login"
                                    minLength={3}
                                    onChange={[Function]}
                                    onInputChange={[Function]}
                                    onSearch={[Function]}
                                    options={Array []}
                                    placeholder="Search for a Github user..."
                                    promptText="Type to search..."
                                    searchText="Searching..."
                                    useCache={true}
                        />
            </FormGroup>
</Col>,
        ],
        "tag": "div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Label
              for="login"
              tag="label"
              widths={
                            Array [
                                          "xs",
                                          "sm",
                                          "md",
                                          "lg",
                                          "xl",
                                        ]
              }
>
              User login
</Label>,
            "md": 2,
            "tag": "div",
            "widths": Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "User login",
              "for": "login",
              "tag": "label",
              "widths": Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ],
            },
            "ref": null,
            "rendered": "User login",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <FormGroup
              tag="div"
>
              <AsyncContainer(OnClickOutside(TypeaheadContainer(WrappedTypeahead)))
                            defaultInputValue="test"
                            delay={200}
                            id="inputsearch"
                            isLoading={false}
                            labelKey="login"
                            minLength={3}
                            onChange={[Function]}
                            onInputChange={[Function]}
                            onSearch={[Function]}
                            options={Array []}
                            placeholder="Search for a Github user..."
                            promptText="Type to search..."
                            searchText="Searching..."
                            useCache={true}
              />
</FormGroup>,
            "md": 10,
            "tag": "div",
            "widths": Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <AsyncContainer(OnClickOutside(TypeaheadContainer(WrappedTypeahead)))
                defaultInputValue="test"
                delay={200}
                id="inputsearch"
                isLoading={false}
                labelKey="login"
                minLength={3}
                onChange={[Function]}
                onInputChange={[Function]}
                onSearch={[Function]}
                options={Array []}
                placeholder="Search for a Github user..."
                promptText="Type to search..."
                searchText="Searching..."
                useCache={true}
/>,
              "tag": "div",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "defaultInputValue": "test",
                "delay": 200,
                "id": "inputsearch",
                "isLoading": false,
                "labelKey": "login",
                "minLength": 3,
                "onChange": [Function],
                "onInputChange": [Function],
                "onSearch": [Function],
                "options": Array [],
                "placeholder": "Search for a Github user...",
                "promptText": "Type to search...",
                "searchText": "Searching...",
                "useCache": true,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
